stock_scheduler_assignation_horizon/models/procurement_group.py

```diff
@@ -2,31 +2,33 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 from datetime import datetime, timedelta
+from logging import getLogger
 
 import pytz
 
 from odoo import models
 from odoo.osv import expression
+from odoo.tools.misc import str2bool
+
+_logger = getLogger(__name__)
 
 
 class ProcurementGroup(models.Model):
     _inherit = "procurement.group"
 
     def _get_moves_to_assign_domain(self, company_id):
         domain = super()._get_moves_to_assign_domain(company_id)
-        company = self.env["res.company"].browse(company_id)
-        if company.is_moves_assignation_limited:
-            domain = self._get_moves_to_assign_domain_assignation_limited(
-                domain, company
-            )
+        global_limit = self.get_global_is_moves_assignation_limited()
+        if global_limit:
+            delay = self.get_global_moves_limit_horizon()
+            domain = self._get_moves_to_assign_domain_assignation_limited(domain, delay)
```

Comment:
While we're in a big refactor, could we combine the domain here instead of inside the function? It would allow for easier override doing this:
```
horizon_domain = self._get_assignation_horizon_domain(delay)
domain = AND([domain, horizon_domain])
```